src/main/java/io/stargate/sgv2/jsonapi/service/cqldriver/CQLSessionCache.java

```diff
@@ -1,3 +1,104 @@
 package io.stargate.sgv2.jsonapi.service.cqldriver;
 
-public class CQLSessionCache {}
+import com.datastax.oss.driver.api.core.CqlSession;
+import com.datastax.oss.driver.api.core.config.DefaultDriverOption;
+import com.datastax.oss.driver.api.core.config.DriverConfigLoader;
+import com.datastax.oss.driver.api.core.config.ProgrammaticDriverConfigLoaderBuilder;
+import com.github.benmanes.caffeine.cache.Cache;
+import com.github.benmanes.caffeine.cache.Caffeine;
+import io.stargate.sgv2.api.common.StargateRequestInfo;
+import io.stargate.sgv2.jsonapi.config.OperationsConfig;
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.inject.Inject;
+import java.nio.file.Path;
+import java.time.Duration;
+import java.util.Objects;
+
+/**
+ * CQL session cache to reuse the session for the same tenant and token. The cache is configured to
+ * expire after <code>CACHE_TTL_SECONDS</code> of inactivity and to have a maximum size of <code>
+ * CACHE_TTL_SECONDS</code> sessions.
+ */
+@ApplicationScoped
+public class CQLSessionCache {
+  /** Configuration for the JSON API operations. */
+  @Inject OperationsConfig operationsConfig;
+
+  /** Stargate request info. */
+  @Inject StargateRequestInfo stargateRequestInfo;
+
+  /** Time to live for CQLSession in cache in seconds. */
+  private static final long CACHE_TTL_SECONDS = 300;
```

Comment:
Since this is time after last use (access) -- I think? -- would this be typical value to use? Or should it be lower, like 60 seconds? (to reduce number of idle sessions).